### PR TITLE
Deduplicate code between new and old generator

### PR DESCRIPTION
**Pull request by oehme**
- Antlr-splitting removed from API surface
- static methods delegate to new generator

### CODE REVIEW

**Comment by oehme** [comment]
@szarnekow We talked about removing the splitting code from the API. Can you make sure this is what you had in mind and merge it?

File: plugins/org.eclipse.xtext.generator/src/org/eclipse/xtext/generator/parser/antlr/AntlrOptions.java [hunk 1]
```diff
@@ -8,100 +8,160 @@
...
-	private boolean backtrack = false;
-	private boolean backtrackLexer = false;
-	private boolean memoize = false;
-	private int k = -1;
-	private boolean ignoreCase = false;
-	private boolean classSplitting = false;
-	private int fieldsPerClass = AntlrParserSplitter.FIELDS_PER_CLASS;
-	private int methodsPerClass = PartialClassExtractor.METHODS_PER_CLASS;
-	private boolean skipUnusedRules = false;
+	org.eclipse.xtext.xtext.generator.parser.antlr.AntlrOptions delegate = new org.eclipse.xtext.xtext.generator.parser.antlr.AntlrOptions();
```
Comment (szarnekow) [inline plugins/org.eclipse.xtext.generator/src/org/eclipse/xtext/generator/parser/antlr/AntlrOptions.java]: Maybe simply extends AntlrOptions?

File: tests/org.eclipse.xtend.core.tests/xtend-gen/org/eclipse/xtend/core/tests/javaconverter/JavaConverterTest.java [hunk 2]
```diff
@@ -638,7 +638,7 @@ public void testOverride() throws Exception {
   
   @Test
   public void testStringLiteralCase() throws Exception {
-    XtendClass xtendClazz = this.toValidXtendClass("class TestStringLiteral { \n\t\t\t\tString withLineWrap=\"string with wrap\\n\";\n\t\t\t\tString str2 = new String(\"\\1\\2\\3\");\n\t\t\t}");
+    XtendClass xtendClazz = this.toValidXtendClass("class TestStringLiteral { \r\n\t\t\t\tString withLineWrap=\"string with wrap\\n\";\r\n\t\t\t\tString str2 = new String(\"\\1\\2\\3\");\r\n\t\t\t}");
```
Comment (szarnekow) [inline tests/org.eclipse.xtend.core.tests/xtend-gen/org/eclipse/xtend/core/tests/javaconverter/JavaConverterTest.java]: Why are the changes in this file necessary?

File: tests/org.eclipse.xtend.ide.tests/xtend-gen/org/eclipse/xtend/ide/tests/contentassist/Bug365368Test.java [hunk 3]
```diff
@@ -19,7 +19,7 @@
   @Test
   public void testEscapedIdentifierNoPrefix() throws Exception {
     ContentAssistProcessorTestBuilder _newBuilder = this.newBuilder();
-    ContentAssistProcessorTestBuilder _append = _newBuilder.append("class C {\n\t\t\tdef void ^as() {}\n\t\t\tdef void aa() {}\n\t\t\tdef void bb() {}\n\t\t\tdef void m() { this.");
+    ContentAssistProcessorTestBuilder _append = _newBuilder.append("class C {\r\n\t\t\tdef void ^as() {}\r\n\t\t\tdef void aa() {}\r\n\t\t\tdef void bb() {}\r\n\t\t\tdef void m() { this.");
```
Comment (szarnekow) [inline tests/org.eclipse.xtend.ide.tests/xtend-gen/org/eclipse/xtend/ide/tests/contentassist/Bug365368Test.java]: Hmm the unrelated changes seem to follow a certain pattern. Is the project configuration broken for xtend.ide.tests?

File: plugins/org.eclipse.xtext.generator/src/org/eclipse/xtext/generator/parser/antlr/AbstractAntlrGeneratorFragment.java [hunk 4]
```diff
@@ -92,23 +92,7 @@ public void addAntlrParam(String param) {
 		String[] result = params.toArray(new String[params.size()]);
 		return result;
 	}
-	
-	private AntlrCodeQualityHelper codeQualityHelper = new AntlrCodeQualityHelper();
```
Comment (szarnekow) [inline plugins/org.eclipse.xtext.generator/src/org/eclipse/xtext/generator/parser/antlr/AbstractAntlrGeneratorFragment.java]: I wanted this to be customizable from mwe without the need to override all Antlr fragments. Even though the old infrastructure will eventually be deprecated, we should not make customizing more painful than necessary.

File: tests/org.eclipse.xtend.ide.tests/xtend-gen/org/eclipse/xtend/ide/tests/contentassist/Bug365368Test.java (hunk 3 above)
Comment (oehme) [inline tests/org.eclipse.xtend.ide.tests/xtend-gen/org/eclipse/xtend/ide/tests/contentassist/Bug365368Test.java]: We generate the system line separator for multi-line Strings. Whenever I do a full build, I have lots of unrelated changes in tests.

Comment (szarnekow) [inline tests/org.eclipse.xtend.ide.tests/xtend-gen/org/eclipse/xtend/ide/tests/contentassist/Bug365368Test.java]: I'm pretty sure the PreferenceStoreWhitespaceInformationProvider is used along the the LineSeparatorHarmonizer. All our projects should have the preference for new text file line delimiter set to unix. If you see unrelated changes of that kind, that's clearly a bug that needs to be addressed.

Comment (oehme) [inline tests/org.eclipse.xtend.ide.tests/xtend-gen/org/eclipse/xtend/ide/tests/contentassist/Bug365368Test.java]: But this is not processed by the LineSeparatorHarmonizer, it's the content of a String. 

It uses what was in the source file. In my case, git converts to windows delimiters using autocrlf. So my source files have Windows delimiters until I check them in, at which point they are converted to Linux delimiters.

Comment (szarnekow) [inline tests/org.eclipse.xtend.ide.tests/xtend-gen/org/eclipse/xtend/ide/tests/contentassist/Bug365368Test.java]: Hmm that's a pretty bad workflow ... isn't there an option to keep the delims as they are in the repo?

Comment (oehme) [inline tests/org.eclipse.xtend.ide.tests/xtend-gen/org/eclipse/xtend/ide/tests/contentassist/Bug365368Test.java]: I used to use the other option (keep them as-is). But Oomph uses autocrlf as the default. That's because it's considered the standard option for git users on Windows.
